data_steward/validation/main.py

```diff
@@ -738,49 +743,58 @@ def list_submitted_bucket_items(folder_bucketitems):
     """
     files_list = []
     object_retention_days = 30
-    object_process_lag_minutes = consts.SUBMISSION_LAG_TIME_MINUTES
     utc_today = datetime.datetime.now(tz=None)
 
     # If any required file missing, stop submission
     folder_bucketitems_basenames = [
         basename(file_name) for file_name in folder_bucketitems
     ]
 
-    if not _has_all_required_files(folder_bucketitems_basenames):
+    to_process_items = [
+        item for item in folder_bucketitems
+        if basename(item) not in resources.IGNORE_LIST
+    ]
+
+    if not to_process_items:
+        return files_list
+
+    # Process if all required files present
+    if _has_all_required_files(folder_bucketitems_basenames):
+        logging.info(f"All required files found, processing.")
+        return to_process_items
+
+    # Check submission times and validate if > 3 hrs old
+    logging.info(
+        f"Delaying processing for hpo_id by 3 hrs (to next cron run) "
+        f"since all required files not present. "
+        f"Missing {set(AOU_REQUIRED_FILES) - set(folder_bucketitems_basenames)}"
+    )
+
+    upper_age_threshold = min(item['timeCreated'] +
+                              datetime.timedelta(days=object_retention_days) -
```

Comment:
How does this `object_retention_days` come into play? I think I am having difficulty understanding what `upper_age_threshold` and `lower_age_threshold` each stand for.